src/wasm-type.h

```diff
@@ -45,6 +45,11 @@ void setTypeSystem(TypeSystem system);
 
 TypeSystem getTypeSystem();
 
+// Dangerous! Frees all types and heap types that have ever been created and
+// resets the type system's internal state. This is only really meant to be used
+// for tests.
+void destroyAllTypes();
```

Comment:
How about `_destroyAllTypes` or `DO_NOT_CALL_destroyAllTypes` or such? That would make it clearer at the callsites (so someone reading through the code that uses this wouldn't get the wrong impression).